### ISSUE TITLE
Please push all tags also to github

### ISSUE THREAD

**Issue by darix** [haproxy/haproxy#2734]
### Your Feature Request

git.haproxy.org is so flaky and unusable a lot of the time that it gets really tedious.

**Comment by chipitsine** [comment]
it is not the answer you are looking for...
but maybe the following versioning info can help [https://www.haproxy.org/download/3.0/src/releases.json](https://www.haproxy.org/download/3.0/src/releases.json) ?

(such a file exists for every supported branch)

**Comment by darix** [comment]
i know when releases happen. but git pull and git clone sometimes do not work for hours. Willy knows about my pain.

**Comment by juliusrickert** [comment]
It appears that dev-build tags, such as `3.2-dev0` to `3.2-dev17`, and the initial release of a branch, e.g., `3.2.0`, are pushed to GitHub. All minor releases, e.g., `3.2.1`, `3.2.2`, are not present.

I have no clue why that is. :D

**Comment by Darlelet** [comment]
haproxy github project only mirrors the master branch from the development version, see https://git.haproxy.org/?p=haproxy.git;a=shortlog;h=refs/heads/master

For actual released haproxy versions, there is a git project for each haproxy version on git.haproxy.org (example haproxy-3.0.git for 3.0 version), and relevant tags are pushed there (in each dedicated repo).

**Comment by pat-s** [comment]
I agree that the current processes result in quite a lot of confusion and makes it hard to have a reliable way to scrape the most recent release to use in some automation tasks.

The JSON mappings are minor-version specific, the github-tags here are missing all patch versions.

An alternative might be the semver-only versions in the DockerHub repo `haproxy`.
